src/components/customers/EditCustomerDocumentLocaleDialog.tsx

```diff
@@ -0,0 +1,162 @@
+import { theme } from '~/styles'
+import DocumentLocales from '~/public/documentLocales.json'
+import { addToast } from '~/core/apolloClient'
+
+gql`
+  fragment EditCustomerDocumentLocale on CustomerDetails {
+    id
+    name
+    externalId
+    billingConfiguration {
+      id
+      documentLocale
+    }
+  }
+
+  mutation updateCustomerDocumentLocale($input: UpdateCustomerInput!) {
+    updateCustomer(input: $input) {
+      id
+      billingConfiguration {
+        id
+        documentLocale
+      }
+    }
+  }
+`
+export interface EditCustomerDocumentLocaleDialogRef extends DialogRef {}
+
+interface EditCustomerDocumentLocaleDialogProps {
+  customer: EditCustomerDocumentLocaleFragment
+}
+
+const documentLocalesData: { value: string; label: string }[] = Object.keys(DocumentLocales).map(
+  (localeKey) => {
+    return {
+      value: localeKey,
+      // @ts-ignore
+      label: DocumentLocales[localeKey],
+    }
+  }
+)
+
+export const EditCustomerDocumentLocaleDialog = forwardRef<
+  DialogRef,
+  EditCustomerDocumentLocaleDialogProps
+>(({ customer }: EditCustomerDocumentLocaleDialogProps, ref) => {
+  const { translate } = useInternationalization()
+  const isEdition = !!customer.billingConfiguration?.documentLocale
+  const [updateDocumentLocale] = useUpdateCustomerDocumentLocaleMutation({
+    onCompleted(res) {
+      if (res.updateCustomer) {
+        addToast({
+          severity: 'success',
+          translateKey: isEdition
+            ? 'text_63ea0f84f400488553caa76f'
+            : 'text_63ea0f84f400488553caa77b',
+        })
+      }
+    },
+  })
+  const formikProps = useFormik<Pick<UpdateCustomerInput, 'billingConfiguration'>>({
+    initialValues: {
+      billingConfiguration: {
+        documentLocale: customer.billingConfiguration?.documentLocale,
+      },
+    },
+    validationSchema: object().shape({
+      billingConfiguration: object().shape({
+        documentLocale: string().required(''),
+      }),
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async (values) => {
+      await updateDocumentLocale({
+        variables: {
+          input: {
+            id: customer.id,
+            ...values,
+            // TODO: Required by API, will fix those non-sense requirements later
+            externalId: customer.externalId,
+            name: customer.name || '',
+          },
+        },
+      })
+    },
+  })
+
+  return (
+    <Dialog
+      ref={ref}
+      title={translate(
+        isEdition ? 'text_63ea0f84f400488553caa65c' : 'text_63ea0f84f400488553caa678'
+      )}
+      description={translate('text_63ea0f84f400488553caa680', { customerName: customer.name })}
+      onClickAway={() => {
+        formikProps.resetForm()
+        formikProps.validateForm()
+      }}
+      actions={({ closeDialog }) => (
+        <>
+          <Button
+            variant="quaternary"
+            onClick={() => {
+              closeDialog()
+              formikProps.resetForm()
```

Comment:
No need to also `validateForm` here ?

Reply:
Form cannot be submit without value selected, so no error will be displayed bellow the input.

So there is no need to validate the form again !